common/lib/xmodule/xmodule/tests/test_content.py

```diff
@@ -173,3 +176,283 @@ def test_static_content_write_js(self):
         js_file_paths = [file_path for file_path in js_file_paths if os.path.basename(file_path).startswith('000-')]
         self.assertEqual(len(js_file_paths), 1)
         self.assertIn("XModule.Descriptor = (function () {", open(js_file_paths[0]).read())
+
+
+@ddt.ddt
+class CanonicalContentTest(SharedModuleStoreTestCase):
+    """
+    Tests the generation of canonical asset URLs for different types
+    of assets: c4x-style, opaque key style, locked, unlocked, CDN
+    set, CDN not set, etc.
+    """
+
+    def setUp(self):
+        super(CanonicalContentTest, self).setUp()
+
+    @classmethod
+    def setUpClass(cls):
+        cls.courses = {}
+
+        super(CanonicalContentTest, cls).setUpClass()
+
+        names_and_prefixes = [(ModuleStoreEnum.Type.split, 'split'), (ModuleStoreEnum.Type.mongo, 'old')]
+        for store, prefix in names_and_prefixes:
+            with cls.store.default_store(store):
+                cls.courses[prefix] = CourseFactory.create(org='a', course='b', run=prefix)
+
+                # Create an unlocked image.
+                unlock_image = Image.new('RGB', (32, 32), 'blue')
+                unlock_buf = StringIO()
+                unlock_image.save(unlock_buf, format='png')
+                unlock_buf.seek(0)
+                unlock_name = '{}_unlock.png'.format(prefix)
+                unlock_key = StaticContent.compute_location(cls.courses[prefix].id, unlock_name)
+                unlock_content = StaticContent(unlock_key, unlock_name, 'image/png', unlock_buf.buf)
+                contentstore().save(unlock_content)
```

Comment:
Maybe extract this image creation into a separate internal method that called five times?